### ISSUE TITLE
The stylelint-disable-line command fails when used within a value list

### ISSUE THREAD

**Issue by Nirazul** [stylelint/stylelint#1628]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where `color-named` reports color keywords like `red` or `green` within sass maps.
This behaviour can't be deactivated, even with stylelint-disable-line comments.

> Which rule, if any, is this issue related to?

`color-named`

> What CSS is needed to reproduce this issue?

```scss
$colors: (
  'error': red,
);
```

as well as all the following disable comments:

```scss
$colors: (
  'error': red, /* stylelint-disable-line color-named */
  'error': red, // stylelint-disable-line color-named
  'error': red, // stylelint-disable-line
  'error': red, /* stylelint-disable-line */
);
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "color-named": "never"
  }
}
```

> Which version of stylelint are you using?

`6.8.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

with `gulp-stylelint`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS maps.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
app/util/color/vars.scss
  52:14  warning  Unexpected named color "red"     color-named
```

**Comment by jeddy3** [comment]
> A bug where color-named reports color keywords like red or green within sass maps.

Checking within SCSS maps is the intended behaviour. The `color-*` rules happen on the *value-level* and so we only return early if the value is not of standard syntax.

> This behaviour can't be deactivated, even with stylelint-disable-line comments.

This is very much a bug, and seemingly not one restricted to SCSS maps. The following standard-syntax CSS also produces a warning.

```css
a {
  background:
    red, /* stylelint-disable-line color-named */
    #000;
}
```

It might be something specific to just this rule (although the maybe more) as the following produces no warnings:

```scss
a {
  color:
    #000
    ,#FFF; /* stylelint-disable-line value-list-comma-space-after */
}
```

I'm going to rename this issue to focus on the `stylelint-disable-line` command and open it out to an investigation. @Nirazul In the meantime, wrapping the whole map or rule in disable commands will work e.g. 

```css
/* stylelint-disable color-named */
$colors: (
  'error': red,
  'warning': yellow
);
/* stylelint-enable */
```

**Comment by jeddy3** [comment]
This *does* produce a warning:

```css
a {
  color:
    #000, /* stylelint-disable-line value-list-comma-space-before */
    #FFF; 
}
```

So, it appears to be something to do with the stylelint-disable-line command being used within a *value list* rather than anything specific to the `color-named`.

**Comment by Nirazul** [comment]
@jeddy3 
It seems that `root.walkComments` in `disableRanges.js` doesn't receive the line comments within a list. So I don't think there's much we can do on our side.

**Comment by m-allanson** [comment]
@Nirazul I agree. 

Here's an example showing how PostCSS parses a comment within a declaration value: https://astexplorer.net/#/6DZiWrYrxa/1/. 

The tree for selector `a` has `comment` and `decl` nodes as expected, but for selector `b` there is only a `decl` node.

**Comment by evilebottnawi** [comment]
@m-allanson yep, see as bug in postcss, need create issue

**Comment by Nirazul** [comment]
As mentioned in the issue, this is by design. Is there another way to get to those inline comments? I figure this might not be the only place where comments are stripped out.

**Comment by evilebottnawi** [comment]
@Nirazul i think be good in postcss v5 go for all node.raws and parse comments and check

**Comment by jeddy3** [comment]
> i think be good in postcss v5 go for all node.raws and parse comments and check

Labelling as "help wanted" in case anyone wants to give this a go.

**Comment by jeddy3** [comment]
Should we just document this limitation in the [configuration doc](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md#turning-rules-off-from-within-your-css) and wait for PostCSS 6.0?
